File: server/src/main/java/message/csmessage/game/CsCharacterState.java
package message.csmessage.game;

import com.fasterxml.jackson.annotation.JsonIgnore;
import message.csmessage.CsMessage;
import session.Session;

public class CsCharacterState extends CsMessage {
    public String id;
    public float locationX;
    public float locationY;
    public float directionX;
    public float directionY;
    public int state;

    public void onMessage(Session session) {
        session.onCharacterState(this);
    }

    @JsonIgnore
    public String getUserId() {
        return id;
    }
}

File: Client/Fruitster/Assets/Script/NetConnection/Messages/CsMessages/GamePlay/CsCharacterState.cs
using System;
using UnityEngine;

namespace Script.Messages.CsMessages
{
    [Serializable]
    public class CsCharacterState : CsMessage
    {
        public string id;
        public float locationX;
        public float locationY;
        public float directionX;
        public float directionY;
        public ObjectState state;

        public CsCharacterState(string id, Vector2 location, Vector2 direction, ObjectState state)
        {
            this.id = id;
            this.locationX = location.x;
            this.locationY = location.y;
            this.directionX = direction.x;
            this.directionY = direction.y;
            this.state = state;
        }
    }
}